src/central-server/admin-service/core/src/main/java/org/niis/xroad/cs/admin/core/service/ConfigurationServiceImpl.java

```diff
@@ -251,6 +331,49 @@ public Set<DistributedFile> getAllConfigurationFiles(int version) {
                 .collect(toSet());
     }
 
+
+    @Override
+    public void uploadConfigurationPart(ConfigurationSourceType sourceType,
```

Comment:
Maybe consider splitting into upload/download services? both service and unit test are reaching 500 lines

Reply:
Extracted Anchor related code to separate method.